pkg/sbom/cyclonedx/marshal.go

```diff
@@ -387,7 +419,9 @@ func (e *Marshaler) reportToCdxComponent(r types.Report) (*cdx.Component, error)
 		properties = appendProperties(properties, PropertyImageID, r.Metadata.ImageID)
 
 		if p.Type == "" {
-			component.BOMRef = e.newUUID().String()
+			// component.BOMRef = e.newUUID().String()
+			// using available values instead of UUID to have consistent report, though the condition is rare
+			component.BOMRef = bomRef(string(r.ArtifactType), r.ArtifactName, r.Metadata.ImageID)
```

Comment:
We might want to check if it complies with the PURL format i.e cyclonedx is valid

Reply:
It doesn't have to a PURL

https://github.com/CycloneDX/specification/blob/c320fc0f0b46873864927d9d5684eea7ba439728/schema/bom-1.5.schema.json#L426C9-L430C11

https://cyclonedx.org/docs/1.5/json/#components_items_bom-ref

Reply:
Ok